### ISSUE TITLE
Fixing seed still leads to differences in validation losses

### ISSUE THREAD

**Issue by sahahner** [ecmwf/anemoi-core#498]
### What happened?

When fixing the seed `ANEMOI_BASE_SEED=71` for two runs using the same config file, we would expect the metrics on mlflow to be similar.

While the [training losses](https://mlflow.ecmwf.int/#/metric?runs=[%22cebfed6f16d8402c8a12cc4df7f4d6dc%22,%220f467ae75c954791856693f0ab9d4d65%22]&metric=%22train_mse_loss_step%22&experiments=[%22346%22]&plot_metric_keys=%5B%22train_mse_loss_step%22%5D&plot_layout={%22autosize%22:true,%22xaxis%22:{%22autorange%22:true,%22type%22:%22linear%22},%22yaxis%22:{}}&x_axis=step&y_axis_scale=linear&line_smoothness=1&show_point=false&deselected_curves=[]&last_linear_y_axis_range=[]) look similar, for some variables the [validation losses](https://mlflow.ecmwf.int/#/metric?runs=[%22cebfed6f16d8402c8a12cc4df7f4d6dc%22,%220f467ae75c954791856693f0ab9d4d65%22]&metric=%22val_mse_loss_epoch%22&experiments=[%22346%22]&plot_metric_keys=%5B%22val_mse_metric%2Fz_500%2F1%22%5D&plot_layout={%22autosize%22:true,%22xaxis%22:{%22autorange%22:true,%22type%22:%22linear%22},%22yaxis%22:{}}&x_axis=step&y_axis_scale=linear&line_smoothness=1&show_point=false&deselected_curves=[]&last_linear_y_axis_range=[]) are very different.

<img width="1143" height="304" alt="Image" src="https://github.com/user-attachments/assets/cd78b6c8-170d-4d21-b5e2-b1a7b153622a" />

### What are the steps to reproduce the bug?

Start a training twice and set the seed to the same one in the jobscript: `export ANEMOI_BASE_SEED=71`

### Version

 training: v0.6.1

### Platform (OS and architecture)

leonardo

### Relevant log output

```shell

```

### Organisation

ECMWF

**Comment by anaprietonem** [comment]
@gabrieloks did you also see the same behaviour in your experiments?
